app/models/services/base-service.ts

```diff
@@ -116,6 +116,12 @@ export default abstract class BaseService<ServiceParamType> {
       const startTime = new Date().getTime();
       logger.info('timing.save-job-to-database.start');
       job = await this._createJob(logger, requestUrl, this.operation.stagingLocation);
+      await job.maybeAttach(db);
+      if (job.attachedStatus.didAttach) {
+        const { originalId, assumedId } = job.attachedStatus;
+        logger.info(`This job attached to a previous job: ${originalId} is now ${assumedId}.`);
+        this.operation.requestId = assumedId;
+      }
```

Comment:
It might be simpler to just return a promise at this point that either resolves to a redirect (if `isAsync` === true) or calls `this_waitForSyncResponse` rather than saving the job that we know is attached and adding the 'mabyeRun' construct below.